src/main/java/org/mskcc/cbio/oncokb/web/rest/slack/ActionId.java

```diff
@@ -56,45 +58,39 @@ public static boolean isEmailAction(ActionId actionId) {
         if (actionId == null) {
             return false;
         }
-        return actionId == SEND_ACADEMIC_FOR_PROFIT_EMAIL
-            || actionId == SEND_ACADEMIC_CLARIFICATION_EMAIL
-            || actionId == SEND_USE_CASE_CLARIFICATION_EMAIL
-            || actionId == SEND_DUPLICATE_USER_CLARIFICATION_EMAIL
-            || actionId == SEND_REGISTRATION_INFO_CLARIFICATION_EMAIL
-            || actionId == SEND_LICENSE_OPTIONS_EMAIL
-            || actionId == SEND_REJECTION_EMAIL
-            || actionId == SEND_REJECT_ALUMNI_ADDRESS_EMAIL;
+        for (MailType mailType : MailType.values()) {
+            if (actionId.equals(mailType.getActionId().orElse(null)))
+                return true;
+        }
+        return false;
     }
 
     public static boolean isConfirmEmailAction(ActionId actionId) {
         if (actionId == null) {
             return false;
         }
-        return actionId == CONFIRM_SEND_ACADEMIC_FOR_PROFIT_EMAIL
-            || actionId == CONFIRM_SEND_ACADEMIC_CLARIFICATION_EMAIL
-            || actionId == CONFIRM_SEND_USE_CASE_CLARIFICATION_EMAIL
-            || actionId == CONFIRM_SEND_DUPLICATE_USER_CLARIFICATION_EMAIL
-            || actionId == CONFIRM_SEND_REGISTRATION_INFO_CLARIFICATION_EMAIL
-            || actionId == CONFIRM_SEND_LICENSE_OPTIONS_EMAIL
-            || actionId == CONFIRM_SEND_REJECTION_EMAIL
-            || actionId == CONFIRM_SEND_REJECT_ALUMNI_ADDRESS_EMAIL;
+        for (MailType mailType : MailType.values()) {
+            if (actionId.equals(mailType.getConfirmActionId().orElse(null)))
+                return true;
+        }
+        return false;
     }
 
     public static boolean isDropdownAction(ActionId actionId) {
         if (actionId == null) {
             return false;
         }
-        return actionId == SEND_ACADEMIC_FOR_PROFIT_EMAIL
-            || actionId == SEND_ACADEMIC_CLARIFICATION_EMAIL
-            || actionId == SEND_USE_CASE_CLARIFICATION_EMAIL
-            || actionId == SEND_DUPLICATE_USER_CLARIFICATION_EMAIL
-            || actionId == SEND_REGISTRATION_INFO_CLARIFICATION_EMAIL
-            || actionId == SEND_LICENSE_OPTIONS_EMAIL
-            || actionId == SEND_REJECTION_EMAIL
-            || actionId == SEND_REJECT_ALUMNI_ADDRESS_EMAIL
-            || actionId == GIVE_TRIAL_ACCESS
-            || actionId == CONVERT_TO_REGULAR_ACCOUNT
-            || actionId == UPDATE_USER
-            || actionId == COLLAPSE;
+
+        // mail actions
+        for (MailType mailType : MailType.values()) {
+            if (actionId.equals(mailType.getActionId().orElse(null)))
+                return true;
+        }
+
+        // non mail actions
+        if (actionId == CONVERT_TO_REGULAR_ACCOUNT || actionId == UPDATE_USER || actionId == COLLAPSE)
```

Comment:
some dropdown actions do not have emails associated with them, so if we wish to add more of those, the logic will still have to be modified in several different classes.

Reply:
TBH, I think we will be in a better place if creating a DropdownOption instead of DropdownEmailOption. Every option will have category, action id, email etc. But it's out of scope of this pr. I will create a separate issue to follow up.